R/reactives.R

```diff
@@ -682,18 +905,26 @@ srcrefToLabel <- function(srcref, defaultLabel) {
 }
 
 #' @export
+format.reactiveExpr <- function(x, ...) {
+  fun <- attr(x, "observable", exact = TRUE)$.origFunc
+  label <- sprintf('reactive(%s)', paste(deparse(body(fun)), collapse='\n'))
+  strsplit(label, "\n")[[1]]
+}
+
+#' @export
 print.reactive <- function(x, ...) {
-  label <- attr(x, "observable", exact = TRUE)$.label
-  cat(label, "\n")
+  cat(paste(format(x), collapse = "\n"), "\n")
 }
 
 #' @export
 #' @rdname reactive
-is.reactive <- function(x) inherits(x, "reactive")
+is.reactive <- function(x) {
+  inherits(x, "reactive")
+}
 
 # Return the number of times that a reactive expression or observer has been run
 execCount <- function(x) {
-  if (is.reactive(x))
+  if (inherits(x, "reactiveExpr"))
```

Comment:
Maybe add `is.reactiveExpr` and `is.reactiveVal` methods? Or maybe `is.reactiveexpr` and `is.reactiveval` to match the capitalization of the existing `is.reactivevalues` method? (yuck)